client/my-sites/checkout/composite-checkout/components/jetpack-pro-redirect-modal/index.tsx

```diff
@@ -0,0 +1,114 @@
+import { Button } from '@automattic/components';
+import { CheckoutCheckIcon } from '@automattic/composite-checkout';
+import { Modal } from '@wordpress/components';
+import { useTranslate } from 'i18n-calypso';
+import JetpackLogo from 'calypso/components/jetpack-logo';
+import { useDispatch, useSelector } from 'calypso/state';
+import { recordTracksEvent } from 'calypso/state/analytics/actions';
+import {
+	JETPACK_DASHBOARD_CHECKOUT_REDIRECT_MODAL_DISMISSED as preferenceName,
+	getJetpackDashboardPreference as getPreference,
+} from 'calypso/state/jetpack-agency-dashboard/selectors';
+import { setPreference } from 'calypso/state/preferences/actions';
+import getPageValueFromUrl from './utils/get-page-value-from-url';
+
+import './style.scss';
+
+interface Props {
+	redirectTo?: string;
+	productSourceFromUrl?: string;
+}
+
+export default function JetpackProRedirectModal( { redirectTo, productSourceFromUrl }: Props ) {
+	const translate = useTranslate();
+	const dispatch = useDispatch();
+
+	const isDismissed = useSelector( ( state ) => getPreference( state, preferenceName ) );
+
+	// Function to set the preference to dismiss the modal and record the event.
+	const dismissAndRecordEvent = () => {
+		dispatch( setPreference( preferenceName, true ) );
+		dispatch( recordTracksEvent( 'jetpack_dashboard_agency_checkout_redirect_modal_dismiss' ) );
+	};
+
+	// Function to record the event when the user clicks on the redirect button.
+	const recordRedirectEvent = () => {
+		dispatch( recordTracksEvent( 'jetpack_dashboard_agency_checkout_redirect_modal_redirect' ) );
+	};
+
+	// Features list of Agency/Pro Dashboard.
+	const features = [
+		translate( 'Up to 60% off our products and bundles.' ),
+		translate( 'A recurring discount (not just for the first year).' ),
+		translate( 'More flexible billing (only pay per day of use, billed monthly).' ),
+		translate( 'Access to our Pro Dashboard – manage all of your sites in one place.' ),
+	];
+
+	const redirectURLPage = getPageValueFromUrl( redirectTo ) ?? '';
+
+	// Show the banner only if the user has not dismissed the banner and
+	// is coming from the Jetpack Plans page or Jetpack page in WP.com
+	if (
+		isDismissed ||
+		! (
+			productSourceFromUrl === 'jetpack-plans' ||
+			[ 'my-jetpack', 'jetpack' ].includes( redirectURLPage )
+		)
+	) {
+		return null;
+	}
+
+	return (
+		<>
+			<Modal
+				open={ true }
+				onRequestClose={ dismissAndRecordEvent }
+				title=""
```

Comment:
Maybe what we need here is **Dialog** instead of **Modal**.
```
import { Dialog } from '@automattic/components';
```

Reply:
I'm not going to make this change here as it requires a lot of changes.